backend/src/test/java/gov/cdc/usds/simplereport/service/DataHubUploaderServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package gov.cdc.usds.simplereport.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.csv.CsvGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import gov.cdc.usds.simplereport.api.model.TestEventExport;
+import gov.cdc.usds.simplereport.db.model.Facility;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.Person;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.AskOnEntrySurvey;
+import gov.cdc.usds.simplereport.db.model.auxiliary.TestResult;
+import gov.cdc.usds.simplereport.test_util.TestDataFactory;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class DataHubUploaderServiceTest extends BaseServiceTest<DataHubUploaderService> {
+  @Autowired private TestDataFactory _dataFactory;
+
+  @Test
+  void serialization() {
+    Organization o = _dataFactory.createValidOrg();
+    Facility f = _dataFactory.createValidFacility(o);
+    AskOnEntrySurvey s =
+        new AskOnEntrySurvey(null, Collections.emptyMap(), null, null, null, null, null, null);
+    TestResult r = TestResult.NEGATIVE;
+    Date d = new Date();
+
+    Person p1 = _dataFactory.createMinimalPerson(o);
+    Person p2 = _dataFactory.createMinimalPerson(o);
+
+    TestEvent te1 = _dataFactory.createTestEvent(p1, f, s, r, d);
+    TestEvent te2 = _dataFactory.createTestEvent(p2, f, s, r, d);
+
+    TestEventExport entry1 = new TestEventExport(te1);
+    TestEventExport entry2 = new TestEventExport(te2);
+
+    CsvMapper mapper = new CsvMapper();
+    mapper
+        .enable(CsvGenerator.Feature.STRICT_CHECK_FOR_QUOTING)
+        .enable(CsvGenerator.Feature.ALWAYS_QUOTE_STRINGS)
+        .enable(CsvGenerator.Feature.ALWAYS_QUOTE_EMPTY_STRINGS);
+
+    CsvSchema schema = mapper.schemaFor(TestEventExport.class).withHeader().withNullValue("\"\"");
+
+    // The data factory generates unique GUIDs for model instances shared by each test event,
+    // so manually overwrite those values to match
+    var date = entry1.getDateResultReleased();
+    var instrumentId1 = entry1.getInstrumentID();
+    var patientId1 = entry1.getPatientId();
+    var resultId1 = entry1.getResultID();
+    var instrumentId2 = entry2.getInstrumentID();
+    var patientId2 = entry2.getPatientId();
+    var resultId2 = entry2.getResultID();
+
+    var serialized =
```

Comment:
**thought (non-blocking):** I worry that this will be slightly annoying to maintain, but tbf it was my idea. And this shouldn't change very often anyhow, so IMO the maintenance cost is worth having a test for it